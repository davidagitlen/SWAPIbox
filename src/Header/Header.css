.Header {
  display: inline;
  flex-direction: row;
  justify-content: space-around;
  height: 300px;
  width: 100%;
  background-color: #000A11;
}

.nav {
  border: none;
  color: gray;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  width: 200px;
  margin: 0px 0px 15px 0px;
  text-decoration: none;
}

.active {
  text-shadow: 0 0 7px yellow;
  text-transform: uppercase;
}

.Header-links {
  background-color: #000A11;
  display: flex;
  flex-direction: row;
  justify-content: space-around;
}

.red {
  color: #FF221E;
  font-weight: bold;
  text-decoration: underline;
}

.yellow {
  color: yellow;
  font-weight: bold;
  text-decoration: underline;
}

.green {
  color: #44CF6C;
  font-weight: bold;
  text-decoration: underline;
}

.blue {
  border: none;
  outline: 0;
  color: #4789CE;
  font-weight: bold;
  text-decoration: underline;
}

.purple{
  color: #9706D9;
  font-weight: bold;
  text-decoration: underline;
}

h1 {
  display: inline-block;
  text-align: center;
  color: yellow;
  font-size: 30px;
  background-color: #000A11;
  width: 100%;
}

img {
  height: 150px;
  width: 150px;
}

.small-header-img {
    padding: 0px 0px 26px 0px;
}

.smaller {
  padding: 23px 28px 34px 28px;
}