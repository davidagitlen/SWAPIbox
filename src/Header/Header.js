import React from 'react';
import { NavLink } from 'react-router-dom';
import PropTypes from 'prop-types';
import xWing from '../images/x-wing.svg';
import chunkyDeathStar from '../images/chunky-death-star.svg';
import bobaFett from '../images/boba-fett.svg';
import nonChunkyHut from '../images/non-chunky-hut.svg';
import nonChunkyFavoriteLightsaber from '../images/non-chunky-favorite-lightsaber.svg';
import './Header.css';

const Header = ({data}) => {
  const favorites = data;
  return (
    <header className='Header'>
      <h1>SWAPIbox</h1>
      <div className='Header-links'>
        <NavLink exact to='/' className='nav' activeClassName='red active'><img className='small-header-img' src={nonChunkyHut} />Home</NavLink>
        <NavLink to='/people' className='nav' activeClassName='yellow active'><img src={bobaFett} />People</NavLink>
        <NavLink to='/planets' className='nav' activeClassName='green active'><img src={chunkyDeathStar} className='deathstar'/>Planets</NavLink>
        <NavLink to='/vehicles' className='nav' activeClassName='blue active'><img src={xWing} />Vehicles</NavLink>
        <NavLink to='/favorites' className='nav' activeClassName='purple active'><img className='small-header-img smaller' src={nonChunkyFavoriteLightsaber} />Favorites | {favorites.length}</NavLink>
      </div>
    </header>
  )
}


export default Header;
  
Header.propTypes = {
  data: PropTypes.array
}