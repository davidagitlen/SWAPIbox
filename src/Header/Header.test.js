import React from 'react';
import { shallow } from 'enzyme';
import { NavLink } from 'react-router-dom';
import Header from './Header';

describe('Header', () => {
  let wrapper, mockPerson;
  
  beforeEach(() => {
    wrapper = shallow(<Header data={[mockPerson]} />);
    mockPerson = { name: 'David', homeworld: 'Kamino', species: 'wookie', population: 20, favorite: 'true' };
  });

  it('should render without crashing', () => {
    expect(wrapper.length).toEqual(1);
  });

  it('should link to the home route when the home navLink is clicked', () => {
    expect(wrapper.find(NavLink).at(0).props().to).toBe('/');
  });

  it('should link to the people route when the people navLink is clicked', () => {
    expect(wrapper.find(NavLink).at(1).props().to).toBe('/people');
  });

  it('should link to the planets route when the planets navLink is clicked', () => {
    expect(wrapper.find(NavLink).at(2).props().to).toBe('/planets');
  });

  it('should link to the vehicles route when the vehicles navLink is clicked', () => {
    expect(wrapper.find(NavLink).at(3).props().to).toBe('/vehicles');
  });

  it('should link to the favorites route when the favorites navLink is clicked', () => {
    expect(wrapper.find(NavLink).at(4).props().to).toBe('/favorites');
  });

  it('should match the snapshot with all data passed in correctly', () => {
    expect(wrapper).toMatchSnapshot();
  })
});

