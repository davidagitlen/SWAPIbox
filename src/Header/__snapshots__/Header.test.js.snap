// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match the snapshot with all data passed in correctly 1`] = `
<header
  className="Header"
>
  <h1>
    SWAPIbox
  </h1>
  <div
    className="Header-links"
  >
    <NavLink
      activeClassName="red active"
      className="nav"
      exact={true}
      to="/"
    >
      <img
        className="small-header-img"
        src="non-chunky-hut.svg"
      />
      Home
    </NavLink>
    <NavLink
      activeClassName="yellow active"
      className="nav"
      to="/people"
    >
      <img
        src="boba-fett.svg"
      />
      People
    </NavLink>
    <NavLink
      activeClassName="green active"
      className="nav"
      to="/planets"
    >
      <img
        className="deathstar"
        src="chunky-death-star.svg"
      />
      Planets
    </NavLink>
    <NavLink
      activeClassName="blue active"
      className="nav"
      to="/vehicles"
    >
      <img
        src="x-wing.svg"
      />
      Vehicles
    </NavLink>
    <NavLink
      activeClassName="purple active"
      className="nav"
      to="/favorites"
    >
      <img
        className="small-header-img smaller"
        src="non-chunky-favorite-lightsaber.svg"
      />
      Favorites | 
      1
    </NavLink>
  </div>
</header>
`;
